miasm2/arch/x86/arch.py

```diff
@@ -712,8 +718,10 @@ def pre_dis(cls, v, mode, offset):
                 pre_dis_info['g1'] = 1
             elif c == '\xf2':
                 pre_dis_info['g1'] = 2
-            elif c == '\xf3':
+            elif c == '\xf3' and v.getbytes(offset + 1) in ['\xa6', '\xa7', '\xae', '\xaf']:
```

Comment:
Here, It may fail in some cases:
in x86, you can have multiple prefix, to the opcode of the instruction may not be just after the current prefix.
To fix this, maybe you can add the test *after* the loop prefix test: if you remove the `c = ''`, `c` will contain the opcode.
Tell me if I am not clear here